device_response.go

```diff
@@ -0,0 +1,106 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package fosite
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+// DeviceResponse represents the device authorization response
+type DeviceResponse struct {
+	Header                  http.Header
+	DeviceCode              string `json:"device_code"`
+	UserCode                string `json:"user_code"`
+	VerificationURI         string `json:"verification_uri"`
+	VerificationURIComplete string `json:"verification_uri_complete,omitempty"`
+	ExpiresIn               int64  `json:"expires_in"`
+	Interval                int    `json:"interval,omitempty"`
+}
+
+// NewDeviceResponse returns a new DeviceResponse
+func NewDeviceResponse() *DeviceResponse {
+	return &DeviceResponse{}
+}
+
+// GetDeviceCode returns the response's device_code
+func (d *DeviceResponse) GetDeviceCode() string {
```

Comment:
no need for these methods now.. since all fields are public.. either unexport all struct fields, or remove these methods..